ethexe/contracts/src/Middleware.sol

```diff
@@ -267,12 +276,50 @@ contract Middleware {
         vaults.remove(vault);
     }
 
-    function getOperatorStakeAt(address operator, uint48 ts)
-        external
-        view
-        _validTimestamp(ts)
-        returns (uint256 stake)
-    {
+    function makeElectionAt(uint48 ts, uint256 maxValidators) public view returns (address[] memory) {
+        require(maxValidators > 0, "Max validators must be greater than zero");
+
+        (address[] memory activeOperators, uint256[] memory stakes) = getActiveOperatorsStakeAt(ts);
+
+        if (activeOperators.length <= maxValidators) {
+            return activeOperators;
+        }
+
+        // Bubble sort descending
```

Comment:
openzeppelin has quicksort:
https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/utils/Arrays.sol

Reply:
I do not see how it can be usable here, because stake and operators must be sort together. Of course we can sort indexes but this will cause additional elements cloning after sorting. Also this is supposed to be off-chain call only